Library/Homebrew/download_strategy.rb

```diff
@@ -330,23 +330,25 @@ def clear_cache
 
   # Private method, can be overridden if needed.
   def _fetch
+    url = @url
+
     if ENV["HOMEBREW_ARTIFACT_DOMAIN"]
-      @url.sub!(%r{^((ht|f)tps?://)?}, ENV["HOMEBREW_ARTIFACT_DOMAIN"].chomp("/") + "/")
-      ohai "Downloading from #{@url}"
+      url.sub!(%r{^((ht|f)tps?://)?}, ENV["HOMEBREW_ARTIFACT_DOMAIN"].chomp("/") + "/")
```

Comment:
Isn't this still indirectly modifying `@url` as `url` is referencing the same `String` instance? If so, this should be fixable by doing `url = @url.dup` or (probably better) by using `String#sub` here.